File: python/oop/consumidor_oop.py
class Consumidor:
    def __init__(self, nome, endereco, cidade):
        self.nome = nome
        self.endereco = endereco
        self.cidade = cidade

    def exibir(self):
        print(f"Nome: {self.nome} | Endereco: {self.endereco} | Cidade: {self.cidade}")

class SistemaCadastro:
    def __init__(self):
        self.consumidores = []

    def cadastrar_consumidor(self):
        nome = input("Nome: ")
        endereco = input("Endereco: ")
        cidade = input("Cidade: ")
        self.consumidores.append(Consumidor(nome, endereco, cidade))
        print("✓ Consumidor cadastrado com sucesso!")

    def listar_consumidores(self):
        if not self.consumidores:
            print("Nenhum consumidor cadastrado.")
            return
        print("\n=== LISTA DE CONSUMIDORES ===")
        for consumidor in self.consumidores:
            consumidor.exibir()
        print("=============================\n")

    def buscar_consumidor(self):
        nome = input("Digite o nome para buscar: ")
        for consumidor in self.consumidores:
            if consumidor.nome.lower() == nome.lower():
                print("\n✓ Consumidor encontrado!")
                consumidor.exibir()
                return
        print("\n✗ Consumidor não encontrado.\n")

    def deletar_consumidor(self):
        nome = input("Digite o nome para deletar: ")
        removidos = [c for c in self.consumidores if c.nome.lower() == nome.lower()]
        self.consumidores = [c for c in self.consumidores if c.nome.lower() != nome.lower()]
        print("✓ Consumidor removido com sucesso!" if removidos else "✗ Consumidor não encontrado.")

    def menu(self):
        while True:
            print("\n=== SISTEMA DE CADASTRO DE CONSUMIDORES (OOP) ===")
            print("1. Cadastrar consumidor")
            print("2. Listar consumidores")
            print("3. Buscar consumidor")
            print("4. Deletar consumidor")
            print("5. Sair")
            opcao = input("Escolha uma opção: ")

            if opcao == '1':
                self.cadastrar_consumidor()
            elif opcao == '2':
                self.listar_consumidores()
            elif opcao == '3':
                self.buscar_consumidor()
            elif opcao == '4':
                self.deletar_consumidor()
            elif opcao == '5':
                print("Encerrando...")
                break
            else:
                print("Opção inválida!")

if __name__ == "__main__":
    SistemaCadastro().menu()

File: java/oop/ConsumidorOOP.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

// Classe Representando o Consumidor
class Consumidor {
    private String nome;
    private String endereco;
    private String cidade;

    public Consumidor(String nome, String endereco, String cidade) {
        this.nome = nome;
        this.endereco = endereco;
        this.cidade = cidade;
    }

    public String getNome() { return nome; }
    public String getEndereco() { return endereco; }
    public String getCidade() { return cidade; }

    public void exibir() {
        System.out.println("Nome: " + nome + " | Endereco: " + endereco + " | Cidade: " + cidade);
    }
}

// Classe Sistema de Cadastro
public class ConsumidorOOP {
    private List<Consumidor> consumidores = new ArrayList<>();
    private Scanner scanner = new Scanner(System.in);

    public void cadastrarConsumidor() {
        System.out.print("Nome: ");
        String nome = scanner.nextLine();
        System.out.print("Endereco: ");
        String endereco = scanner.nextLine();
        System.out.print("Cidade: ");
        String cidade = scanner.nextLine();
        consumidores.add(new Consumidor(nome, endereco, cidade));
        System.out.println("✓ Consumidor cadastrado com sucesso!");
    }

    public void listarConsumidores() {
        if (consumidores.isEmpty()) {
            System.out.println("Nenhum consumidor cadastrado.");
            return;
        }
        System.out.println("\n=== LISTA DE CONSUMIDORES ===");
        consumidores.forEach(Consumidor::exibir);
        System.out.println("=============================\n");
    }

    public void buscarConsumidor() {
        System.out.print("Digite o nome para buscar: ");
        String nome = scanner.nextLine();
        for (Consumidor c : consumidores) {
            if (c.getNome().equalsIgnoreCase(nome)) {
                System.out.println("\n✓ Consumidor encontrado!");
                c.exibir();
                return;
            }
        }
        System.out.println("\n✗ Consumidor não encontrado.\n");
    }

    public void deletarConsumidor() {
        System.out.print("Digite o nome para deletar: ");
        String nome = scanner.nextLine();
        consumidores.removeIf(c -> c.getNome().equalsIgnoreCase(nome));
        System.out.println("Operação concluída. (Caso o nome existisse, foi removido)");
    }

    public void menu() {
        while (true) {
            System.out.println("\n=== SISTEMA DE CADASTRO DE CONSUMIDORES (OOP) ===");
            System.out.println("1. Cadastrar consumidor");
            System.out.println("2. Listar consumidores");
            System.out.println("3. Buscar consumidor");
            System.out.println("4. Deletar consumidor");
            System.out.println("5. Sair");
            System.out.print("Escolha uma opção: ");
            int opcao = scanner.nextInt();
            scanner.nextLine();

            switch (opcao) {
                case 1:
                    cadastrarConsumidor();
                    break;
                case 2:
                    listarConsumidores();
                    break;
                case 3:
                    buscarConsumidor();
                    break;
                case 4:
                    deletarConsumidor();
                    break;
                case 5:
                    System.out.println("Encerrando...");
                    scanner.close();
                    return;
                default:
                    System.out.println("Opção inválida!");
            }
        }
    }

    public static void main(String[] args) {
        new ConsumidorOOP().menu();
    }
}
